R/event_manager.R

```diff
@@ -90,6 +125,42 @@ EventManager <- R6Class(
     # notifications, others do not. (Not really sure why.)
     event_enable_domains = NULL,
 
+    event_domain_manually_enabled = list(),
+
+    maybe_event_domain_enable = function(domain) {
```

Comment:
It would be clearer with names like `maybe_auto_event_domain_enable()` , and `maybe_auto_event_domain_disable()` -- otherwise I have to read through all the logic to understand why it's a maybe.